Marlin/src/feature/backlash.cpp

```diff
@@ -121,6 +121,37 @@ void Backlash::add_correction_steps(const int32_t &da, const int32_t &db, const
       // This correction reduces the residual error and adds block steps
       if (error_correction) {
         block->steps[axis] += ABS(error_correction);
+
+        // Recalculate block->millimeters so that acceleration is calculated correctly
+        // (Relocated from module/planner.cpp)
+        if (true NUM_AXIS_GANG(
+          && block->steps.a < MIN_STEPS_PER_SEGMENT, && block->steps.b < MIN_STEPS_PER_SEGMENT, && block->steps.c < MIN_STEPS_PER_SEGMENT,
+          && block->steps.i < MIN_STEPS_PER_SEGMENT, && block->steps.j < MIN_STEPS_PER_SEGMENT, && block->steps.k < MIN_STEPS_PER_SEGMENT,
+          && block->steps.u < MIN_STEPS_PER_SEGMENT, && block->steps.v < MIN_STEPS_PER_SEGMENT, && block->steps.w < MIN_STEPS_PER_SEGMENT)
+        ) {
+          block->millimeters = TERN0(HAS_EXTRUDERS, ABS(dist_mm.e));
+        }
+        else if (hints.millimeters) {
+          block->millimeters = hints.millimeters;
+        }
+        else {
+          const xyze_pos_t displacement = LOGICAL_AXIS_ARRAY(
+            dist_mm.e,
+            #if ANY(CORE_IS_XY, MARKFORGED_XY, MARKFORGED_YX)
+              dist_mm.head.x, dist_mm.head.y, dist_mm.z,
+            #elif CORE_IS_XZ
+              dist_mm.head.x, dist_mm.y, dist_mm.head.z,
+            #elif CORE_IS_YZ
+              dist_mm.x, dist_mm.head.y, dist_mm.head.z,
+            #else
+              dist_mm.x, dist_mm.y, dist_mm.z,
+            #endif
+            dist_mm.i, dist_mm.j, dist_mm.k,
+            dist_mm.u, dist_mm.v, dist_mm.w
+          );
+          block->millimeters = get_move_distance(displacement OPTARG(HAS_ROTATIONAL_AXES, cartesian_move));
+        }
+
```

Comment:
Well this obviously can't just be moved here, as most of the variables it references only exist in its original location. If the aim is to deal with `block->steps[axis]` having been affected by `error_correction` then it would be better to make a method that returns all the axis `error_correction` values and then apply those to the logic in `Planner::_populate_block`.

Reply:
My apologies, rookie git use, this wasn't supposed to end up in the merge request. 
I'll cancel this PR until I've finished working on the backlash issue